regression/contracts/invar_check_pointer_modifies-01/main.c

```diff
@@ -0,0 +1,18 @@
+#include <assert.h>
+#include <stdlib.h>
+
```

Comment:
Random location for comment: would you mind merging the second commit (the regression test) into the first one? When later looking at git history, this makes it much easier to understand why a particular change was made.

Reply:
Makes sense, thanks for the quick feedback!